### PR TITLE
Add integration tests for job controller

### PR DESCRIPTION
**Pull request by kerthcet**
Signed-off-by: Kante Yin <kerthcet@gmail.com>

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

~The e2e test flow:~
~1. Create a job with no matched localQueue, job is suspend.~
~2. Create a localQueue but located in another namespace with job, job is suspend.~
~3. Unsuspend the job manually, back to suspend status.~
~4. Create a localQueue located in the same namespace. job is unsuspend.~
~5. Delete the job, workload is also deleted.~

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes-sigs/kueue/issues/546

### CODE REVIEW

**Comment by kerthcet** [comment]
Should be merged after https://github.com/kubernetes-sigs/kueue/pull/550
/hold

**Comment by alculquicondor** [comment]
can you do this in integration?

File: test/integration/controller/job/job_controller_test.go [hunk 1]
```diff
@@ -652,4 +652,43 @@ var _ = ginkgo.Describe("Job controller interacting with scheduler", func() {
...
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "e2e-",
+			},
+		}
+		gomega.Expect(k8sClient.Create(ctx, ns2)).To(gomega.Succeed())
+		defer func() {
+			gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns2)).To(gomega.Succeed())
+		}()
+
+		ginkgo.By("create a localQueue located in a different namespace with the job")
```
Comment (alculquicondor) [inline test/integration/controller/job/job_controller_test.go:668]:
```suggestion
		ginkgo.By("create a localQueue located in a different namespace as the job")
```

File: test/integration/controller/job/job_controller_test.go [hunk 2]
```diff
@@ -652,4 +652,43 @@ var _ = ginkgo.Describe("Job controller interacting with scheduler", func() {
...
+
+		ginkgo.By("job should be suspend")
+		lookupKey := types.NamespacedName{Name: prodJob.Name, Namespace: prodJob.Namespace}
+		createdProdJob := &batchv1.Job{}
+		gomega.Eventually(func() *bool {
+			gomega.Expect(k8sClient.Get(ctx, lookupKey, createdProdJob)).Should(gomega.Succeed())
+			return createdProdJob.Spec.Suspend
+		}, util.Timeout, util.Interval).Should(gomega.Equal(pointer.Bool(true)))
+
+		ginkgo.By("creating another localQueue of the same name and in the same namespace with job")
```
Comment (alculquicondor) [inline test/integration/controller/job/job_controller_test.go:684]:
```suggestion
		ginkgo.By("creating another localQueue of the same name and in the same namespace as the job")
```

File: test/integration/controller/job/job_controller_test.go [hunk 3]
```diff
@@ -652,4 +652,43 @@ var _ = ginkgo.Describe("Job controller interacting with scheduler", func() {
...
+		prodLocalQ = testing.MakeLocalQueue(localQueue.Name, ns.Name).ClusterQueue(prodClusterQ.Name).Obj()
+		gomega.Expect(k8sClient.Create(ctx, prodLocalQ)).Should(gomega.Succeed())
+
+		ginkgo.By("job should be unsuspend")
```
Comment (alculquicondor) [inline test/integration/controller/job/job_controller_test.go:688]:
```suggestion
		ginkgo.By("job should be unsuspended")
```

**Review by alculquicondor — COMMENTED** [review]
just nits
/approve

File: test/integration/controller/job/job_controller_test.go (hunk 3 above)
Comment (kerthcet) [reply test/integration/controller/job/job_controller_test.go]: donw

**Comment by kerthcet** [comment]
squashed and force pushed.

**Comment by alculquicondor** [comment]
/lgtm

can you update the PR description?

**Comment by kerthcet** [comment]
> /lgtm
> 
> can you update the PR description?

updated.

**Pull request merged**